Dockerfile

```diff
@@ -1,36 +1,86 @@
-#Dockerfile created following this blog post: https://medium.com/@albertazzir/blazing-fast-python-docker-builds-with-poetry-a78a66f5aed0
 #PYTHON image
 # Use the official Docker Python image because it has the absolute latest bugfix version of Python
 # it has the absolute latest system packages
 # it’s based on Debian Bookworm (Debian 12), released June 2023
-# Image size is 51MB
+# Initial Image size is 51MB
+# At the end Image size is 156MB
 
 #I did not recommed to use alpine image because it lacks the package installer pip and the support for installing
 #wheel packages, which are both needed for installing applications like Pandas and Numpy.
-FROM python:3.11-slim-bookworm as builder
 
-RUN pip install poetry==1.5.1
+# The base layer will contain the dependencies shared by the other layers
+FROM python:3.11-slim-bookworm as base
 
-ENV POETRY_NO_INTERACTION=1 \
-    #POETRY_VIRTUALENVS_IN_PROJECT=1 \
-    #POETRY_VIRTUALENVS_CREATE=1 \
-    POETRY_CACHE_DIR=/tmp/poetry_cache
+# Create a system group named "user" with the -r flag
+RUN groupadd -r app_user \
+    # Create a system user named "user" with the -r flag
+    && useradd -r -g app_user app_user
 
+# Set the working directory to /app
 WORKDIR /app
 
-COPY pyproject.toml poetry.lock ./
+# Change the ownership of the working directory to the non-root user "user"
+RUN chown -R app_user:app_user /app
 
-# Avoid installing development dependencies with poetry install --without dev , as you won’t need linters and tests suites in your production environment.
-#To avoid poetry re-install the dependencies use the --no-root option, which instructs Poetry to avoid installing the current project into the virtual environment.
-RUN --mount=type=cache,target=$POETRY_CACHE_DIR poetry install --without dev --no-root
-
-FROM python:3.11-slim-bookworm as runtime
+# Switch to the non-root user "user"
+USER app_user
 
+#Use this argument to install development or production dependencies
+#true = development / false = production
+ARG DEV=true
 #ENV VIRTUAL_ENV=/app/.venv \
```

Comment:
I left the lines related to virtual environment because I would like to understand why it is not working inside the docker but I have not time right now to do this check

Reply:
With K'Ron's support I've learned how to manage virtual environment inside the docker. I was creating the virtual environment, but I was not activated it. As each line in the docker is a layer, if we create create the virtual environment in one line and want to use it in another line it won't work. We should do `RUN . /app/venv/bin/activate && python main.py`. Right now, our docker use virtual environment and poetry dependencies are separated from the project dependencies